packages/orama/src/trees/avl.ts

```diff
@@ -224,32 +224,31 @@ function getNodeByKey<K, V>(node: Node<K, V> | null, key: K): Node<K, V> | null
       return node;
     }
   }
-  return null;
 }
 
 export function create<K, V>(key: K, value: V): Node<K, V> {
   return {
     key,
     value,
-    left: null,
-    right: null,
+    left: undefined,
+    right: undefined,
     height: 0,
   }
 }
 
 export function insert<K, V>(root: Node<K, V>, key: K, value: V): Node<K, V> {
-  let parent = null;
+  let parent;
   let current = root;
 
-  while (current !== null) {
+  while (current !== undefined) {
     parent = current;
     if (key < current.key) {
         current = current.left as Node<K, V>;
     } else if (key > current.key) {
         current = current.right as Node<K, V>;
     } else {
       // assuming value is an array here
-      (current.value as string[]) = (current.value as string[]).concat(value as string);
+      (current.value as string[]).push(...value as string);
```

Comment:
This will not lead in something like this?

![image](https://github.com/oramasearch/orama/assets/12551007/0487d666-e999-41dc-b437-cb1e65787639)

Reply:
You are correct, I need to remoce the spread operator